mllib/src/main/scala/org/apache/spark/ml/tuning/TrainValidationSplit.scala

```diff
@@ -198,6 +202,11 @@ class TrainValidationSplitModel private[ml] (
     @Since("1.5.0") val validationMetrics: Array[Double])
   extends Model[TrainValidationSplitModel] with TrainValidationSplitParams with MLWritable {
 
+  /** A Java/Python-friendly auxiliary constructor. */
```

Comment:
If this is intended to be Java friendly then we probably don't want to make it private to ML scope and we should expand on the javadoc. If it is just for Python it seems this would probably be better suited to PythonMLLibAPI.scala